presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportServlet.java

```diff
@@ -128,12 +128,14 @@ protected void parseURI(String requestURI, HttpServletRequest request, HttpServl
         OutputBufferId bufferId = null;
         long token = 0;
 
+        String sanitizedRequestURI = requestURI.replaceAll("[\\r\\n]", "");
```

Comment:
So this removes \r and \n from the URI. But having these chars would not be a valid URI in the first place. The server listener doesn't filter this out already?
The vulnerability talks about having these values in the header - because these values are used to separate header and body in HTTP.

Validating the input is fine. Just surprised this isn't already taken care of.

Reply:
+1. 
We need to sanitize the request headers in order to handle this vulnerability.

Reply:
So just to clarify, you are suggesting I get all headers from the request variable and sanitize each one using the same method here?